apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/status/AnnotationMatchExceptionCheckStrategy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.context.status;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+
+public class AnnotationMatchExceptionCheckStrategy implements ExceptionCheckStrategy {
+
+    private final Set<Class<? extends Throwable>> ignoredExceptions = new CopyOnWriteArraySet<>();
```

Comment:
@kezhenxu94 @EvanLjp Any of you know the performance difference between `CopyOnWriteArraySet` and `ConcurrentHashMap`? This check is highly performance sensitive.

Reply:
Read some doc , and replace it  with the  set view of concurrentHashMap to store exception class.